torch/_decomp/decompositions.py

```diff
@@ -2957,6 +2957,103 @@ def _sum_tensors(ts: Iterable[Tensor]) -> Tensor:
     return reduce(torch.add, ts)
 
 
+def _linspace_from_neg_one(
+    num_steps: int, align_corners: bool, dtype: torch.dtype, device: torch.device
+):
+    if num_steps <= 1:
+        return torch.tensor(0, device=device, dtype=dtype)
+
+    a = ((num_steps - 1) / num_steps) if not align_corners else 1
+    return torch.linspace(-a, a, steps=num_steps, device=device, dtype=dtype)
+
+
+def _make_base_grid_4d(theta: Tensor, n: int, h: int, w: int, align_corners: bool):
+    dtype = theta.dtype
+    device = theta.device
+
+    # Using padding and summation generates a single kernel vs using torch.stack where 3 kernels generated
+    # corresponding to each individual tensor: grid_x, grid_y, grid_one
+    grid_x = (
+        _linspace_from_neg_one(w, align_corners, dtype, device)
+        .view(1, 1, w, 1)
+        .expand(n, h, w, 1)
+    )
+    grid_y = (
+        _linspace_from_neg_one(h, align_corners, dtype, device)
+        .view(1, h, 1, 1)
+        .expand(n, h, w, 1)
+    )
+    grid_one = (
+        torch.tensor(1, dtype=dtype, device=device).view(1, 1, 1, 1).expand(n, h, w, 1)
+    )
+
+    grid_x = torch.nn.functional.pad(grid_x, pad=(0, 2), mode="constant", value=0)
```

Comment:
Could you leave a comment here and in the other path saying that this is just a hack, and that we should user `torch.stack` here once https://github.com/pytorch/pytorch/pull/104480 is merged?